### DISCUSSION TITLE
Backspace, Home and Delete keypresses (possibly others) applied to wrong file in Visual Studio with Copilot 1.94.0.0 installed.

### DISCUSSION THREAD

**Discussion by JocelynSachs** [community/community#61730] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Several times now since installing Copilot I've been typing code, pressed backspace to correct a mistake - and nothing seems to happen.

I just had it happen again now. Hammering backspace and delete seemed to do nothing, but I could still type new code.

I then clicked on another open file in VS, and found that all the backspacing and deleting I had been doing had been applied to that file instead, at the place where I had left the cursor.

**Comment by rccnw** [comment]
Here it is March 2024 and this bug is still happening. Anybody home?  This is pretty hard to take. I had no idea the keys were being applied to another file till I read this bug report from 8 months ago.    Using GitHub Copilot extension 1.156.0.0 in VS2022 Community

**Reply by thomasmaisel** [reply]
I had the same problem. Backspace did not work in Copilot Chat Toolwindow, as well in the Inline Chat.
I'm also a VSVim user, but I'm not sure if it's related to the extension since uninstalling the extension did not fix it.

I fixed it by:
- exporting my current keyboard bindings
- resetting the keybindings in the options menu (now, backspace works again)
- edit the exported file. delete the line: 
    `<RemoveShortcut Command="Edit.DeleteBackwards" Scope="Text Editor">Bkspce</RemoveShortcut>`
- import the keyboard bindings again.

**Comment by ayim** [comment]
Yikes. @rccnw @JocelynSachs please reach out to me directly at aayim@microsoft.com so that we can schedule a screenshare and collect logs.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
github: ayim | aayim@microsoft.com

**Reply by rccnw** [reply]
Sent email, thanks.

**Comment by cdrenshaw** [comment]
I have the same issue or similar issue... I have the Copilot Chat window docked in Visual Studio.  The issue always occurs when I open the window, pin it, and then click Create New Thread.  If I have to backspace while typing, it backspaces in whatever open document my cursor was last in which deletes some of my code from wherever the cursor was at.  I have to unpin the window and close it, then reopen it and pin it again for backspace to work correctly.

![image](https://github.com/community/community/assets/39387442/b9ac5831-e93d-4d4f-8041-4ffc23178651)

**Reply by ayim** [reply]
@cdrenshaw please reach out to me via email at aayim@microsoft.com so that we can do a debugging session together.
Let me know your availability early next week.

-A

**Reply by cdrenshaw** [reply]
Sent.  Thanks!

**Reply by carlos-quintero** [reply]
It happens to me too quite regularly, in the GitHub Copilot Chat normal keys are recognized and text is written, but the backspace key goes to the code window and deletes actual code without noticing it at first glance.

**Reply by ayim** [reply]
@carlos-quintero we've found the error and in the middle of releasing a fix.
If you're on Visual Studio Preview (17.10 Preview 2 or above), the fix is already in.
We're investigating backporting the fix to 17.9 and 17.8

**Reply by jbalinski** [reply]
Would also like to see the fix pushed out sooner than later. Copilot chat is near unusable with this issue. I updated to Visual Studio 17.9.7 hoping that it might be fixed but no luck.

**Reply by SJFriedl** [reply]
I'm not sure if I'm having the same problem, but in the latest VS 17.10.4, opening a GitHub Copilot chat window, backspace isn't working for me either nor was it in 17.10.3 that I just updated from. If my backspace is being routed to some other window, I haven't been able to see which: it's just swallowed as far as I can tell.

It behaves the same way whether the window is floating or docked. I'm on the latest Windows 11.

I use ReSharper and VsVim extensions, if it matters, but the problem persists even after I've disabled both of them in varying combinations.

Back to ChatGPT for me, I guess.

**Reply by ayim** [reply]
@SJFriedl is this happening when you're editing in the chat window or when you're editing grey text in the main editor window?

**Reply by SJFriedl** [reply]
This is in the little GitHub Copilot Chat window. Typing works fine, left-arrow does move left, but backspace doesn't do anything apparent.
 
![image](https://github.com/user-attachments/assets/434a4232-893e-46e8-8a2a-5c6c9c5f9a84)

**Reply by noellelc** [reply]
Do you have Visual Assist by chance? I know that was also affecting some folks using the chat window, but updating that extension seemed to resolve the issue.

**Reply by SJFriedl** [reply]
No, I've never used this extension.

To try to rule things out, I also have VS installed on my laptop - Windows 11 - and even after disabling **every** extension, I get the same behavior in the Copilot chat window (i.e., backspace doesn't appear to do anything).

The lappy has far less stuff on it so is not polluted like my daily driver desktop is.

Neither machine has any non-Microsoft desktop security software on it. Both use AMD processors, if it matters.

**Reply by rubenseggers** [reply]
I had the same problem and and fixed it using: Tools > Import and export settings... > Reset all settings. 

It gives you an option to save your current settings, and you can import a selection of those saved settings. Repeatedly resetting and loading half of the settings—always nice to find a good use case for a classic O(log N) binary search—showed the culprit was: All settings > Options > Environment > Keyboard.

So saving and resetting the settings, subsequently importing all but the "Keyboard" one, did the trick for me. I had to reconfigure some keyboard shortcuts I was using, but other than that all my previous settings are working just fine. Hope this helps!

**Reply by SJFriedl** [reply]
Hmm, this did fix the backspace issue on my laptop (thank you!), but broke a bunch of other key bindings.

This suggests there's really no way to avoid the maze of twisty passages of keyboard bindings to make things work the way I want it, and this is completely overwhelming.  Not willing to risk destroying my main dev system key bindings, so will just stick with ChatGPT for AI help.

Appreciate your sharing a solution.

**Reply by SJFriedl** [reply]
**Update**: I realized that I could save all my settings in an XML file (which would allow me to recover in case of disaster), and ran this on my primary development desktop: it worked, and after a brief click on a VsVim setting (allow VsVim to handle all), this appears to have completely resolved this backspace issue on both of my VS2022 installs.

I'm grateful for the guidance from this thread, and hope that the feedback is helpful to the Copilot team.

**Reply by rubenseggers** [reply]
Interesting, I'm also using VsVim, but I actually do want some of those shortcuts (in the VsVim -> Keyboard menu) to be handled by VS. However, it's good to know that VsVim key bindings are part of the problem, might this or similar errors occur in the future. Thanks for your follow up and all the best.

**Reply by SJFriedl** [reply]
I'm not 100% sure that VsVim is implicated here: the backspace was eaten even when VsVim (and ReSharper) were completely disabled.

It's possible that VsVim interacts with the key bindings in a way that lingered even after being disabled, being fixed by the "Reset All Settings" move.  I looked through the settings XML file of settings prior to reset, and didn't see anything obvious.
